iothub/device/tests/ClientPropertyCollectionTests.cs

```diff
@@ -0,0 +1,140 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.Azure.Devices.Client;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Azure.Devices.Shared;
+
+namespace Microsoft.Azure.Devices.Client.Tests
+{
+    [TestClass]
+    [TestCategory("Unit")]
+    public class ClientPropertyCollectionTests
+    {
+        private const string propertyName = "propertyOne";
+        private const string simplePropertyValue = "propertyValue";
+        private const string componentName = "testableComponent";
+        private const string writablePropertyDescription = "testableWritablePropertyDescription";
+        private const string updatedPropertyValue = "updatedPropertyValue";
+
+        [TestMethod]
+        public void ClientPropertyCollection_CanAddSimpleObjectAndGetBackWithoutDeviceClient()
```

Comment:
similar tests for other primitive types, complex objects, arrays, nested objects etc.
We could use parameterized tests here, unit tests don't take long, so running them serially won't have much of an impact.